core/src/hir/type_context.rs

```diff
@@ -1,9 +1,19 @@
 //! Store all the types contained in the HIR.
 
-use super::{EnumDef, OpaqueDef, OutStructDef, StructDef};
+use super::{
```

Comment:
nit: worth splitting into lowering.rs and type_context.rs, with all the individual lowering functions in lowering.rs